00_Base/src/services/credentials.service.ts

```diff
@@ -546,15 +662,9 @@ export class CredentialsService {
     versionNumber: VersionNumber,
     credentials: CredentialsDTO,
   ): Promise<ClientVersion> {
-    const existingVersionDetails = clientInformation.clientVersionDetails.find(
-      (v: ClientVersion) => v.version === versionNumber,
-    );
-    if (!existingVersionDetails) {
-      throw new NotFoundError('Version details not found');
-    }
     const clientVersion = await this.getVersionDetails(
       versionNumber,
-      existingVersionDetails.url,
+      credentials.url,
```

Comment:
this method is getting VersionDetails--the credentials.url gets Versions
the VersionDetail url is in the Version object, which is why the previous lookup was happening

Reply:
Hi Thana, thanks for the review! 

This `getVersionDetails` method actually get versions at first and then get version details. The input url is used to get version so I changed it to use this `credentials.url `. 

I also removed the previous lookup for `existingVersionDetails` because when we initially receives the postCredential call from MSP, there has not been client side version details. We need to use the credential url from the request body to get client side versions and then details (we do it inside this `getVersionDetails` method). The previous lookup and throw exceptions make the process fail.

Reply:
yep, the one helper `this.getVersionDetails(` is designed to return VersionDetails which is created after performing both of the get versions and the get version details requests

Reply:
oh wait yeah @lydiazcheng, Thana is right, as I just now pointed out, you need to pass in the versions endpoint url here and not the credentials endpoint because that function will perform both GET versions requests

Reply:
Hi @elliot-sabitov, the passed in url is only used to get versions. The `baseUrl` is then replaced by the real version url later after we receive the response of get versions (this [line](https://github.com/citrineos/citrineos-ocpi/blob/017d7d8093d44aa4b74f8461bac75bc0ef056224/00_Base/src/services/credentials.service.ts#L585)). 

And we are not able to pass in any version endpoint url into this `getVersionDetails` because we don't have any client side `existingVersionDetails` at the moment when calling this `getVersionDetails`. As described in OCPI Registration process, when we play as a receiver, the MSP send us postCredential request, then we start to send get version and get the version details request to get client side version details.

Reply:
ah I see right okay, makes sense